src/dotty/tools/dotc/core/SymDenotations.scala

```diff
@@ -56,6 +56,7 @@ trait SymDenotations { this: Context =>
     stillValid(owner) && (
       !owner.isClass
       || owner.isRefinementClass
+      || owner.is(Scala2x)
```

Comment:
This reminds me of https://github.com/lampepfl/dotty/commit/4b0cc8aabf0dbbd74979faedee7371c41177b755 which fixed an issue with stale Java symbol and outer fields, are you sure that the stale outer field here is not incorrectly generated by Dotty?